storage-fetch/src/download.rs

```diff
@@ -10,25 +11,36 @@ use cid::Cid;
 use futures::StreamExt;
 use ipld_core::codec::Codec;
 use ipld_dagpb::{DagPbCodec, PbNode};
-use libp2p::{Multiaddr, PeerId, Swarm};
+use libp2p::{
+    noise,
+    request_response::{self, Message, ProtocolSupport},
+    tcp, yamux, Multiaddr, PeerId, StreamProtocol, Swarm, SwarmBuilder,
+};
 use libp2p_core::ConnectedPoint;
-use libp2p_swarm::{ConnectionId, DialError, SwarmEvent};
+use libp2p_swarm::{ConnectionId, DialError, NetworkBehaviour, SwarmEvent};
 use mater::{blockstore::ReadWriteBlockstore, FileReader, DAG_PB_CODE, RAW_CODE};
+use primitives::p2p::{PieceInfoRequest, PieceInfoResponse, SP_REQUEST_RESPONSE_PROTOCOL};
 use thiserror::Error;
 use tokio::{
     fs::{File, OpenOptions},
     io::AsyncSeekExt,
 };
 use tracing::{debug, error, info, instrument, trace};
 
-use crate::p2p::{new_swarm, Behaviour, BehaviourEvent, InitSwarmError};
+const MAX_MULTIHASH_LENGTH: usize = 64;
 
 /// Errors that can occur while retrieving some content.
 #[derive(Debug, Error)]
-pub enum ClientError {
-    /// Error occurred while initialing swarm
-    #[error("Swarm initialization error: {0}")]
-    InitSwarm(#[from] InitSwarmError),
+pub enum DownloadError {
+    /// Is used when the storage provider doesn't have a piece requested
+    #[error("Unknown piece")]
+    PieceUnknown,
```

Comment:
This error doesn't seem to match the doc. Shouldn't it be `NoPieceRequested`? Or maybe the doc should instead state `When an unknown piece is requested` or something?